src/ripple/app/misc/ValidatorList.h

```diff
@@ -343,7 +343,7 @@ class ValidatorList
     */
     bool
     load(
-        PublicKey const& localSigningKey,
+        std::optional<PublicKey const> localSigningKey,
```

Comment:
Should be ` std::optional<PublicKey> const&`

Reply:
ok. I get that the `const&` helps avoid the extra copy. Shouldn't it be `std::optional<PublicKey const> const&` to ensure that the contents of the optional is not modified?

I don't know how the optional works, but if it stores a pointer to the PublicKey object, then your suggestion could still violate the `const-` ness. On the other hand, if an entire copy of the `PublicKey` object is stored within the optional, then your version works just fine.

Reply:
According to https://en.cppreference.com/w/cpp/utility/optional/operator*, the `const` overloads of the `->` and `*` operators return a const ref. The same is true for https://en.cppreference.com/w/cpp/utility/optional/value. So it would not be easy to modify the value. (I almost never say "impossible", because you can always do `const_cast` craziness and such.)

Reply:
okay, I'll go through these references, thanks